compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -2500,20 +2535,24 @@ fn show_candidates(
     }
 
     if !accessible_path_strings.is_empty() {
-        let (determiner, kind, name) = if accessible_path_strings.len() == 1 {
-            ("this", accessible_path_strings[0].1, format!(" `{}`", accessible_path_strings[0].0))
+        let (determiner, kind, name, post) = if accessible_path_strings.len() == 1 {
```

Comment:
Maybe can use
```suggestion
        let (determiner, kind, name, post) = if let [first_element] = accessible_path_strings {
```
form here, should look better.